src/comm/client/client.js

```diff
@@ -96,8 +98,38 @@ class Client{
     /**
     * Initialise client object
     * @return {Promise} promise object
+    * @param {JSON} demo object which trakcs real time metrics
+    * @param {String} config path of the configuration file
+    * @param {String} absCaliperDir caliper directory
+    * @param {JSON} listener_child child process
+    * @param {String} networkFile network file path
+    * @param {JSON} t test object
     */
-    async init() {
+    async init(demo, config, absCaliperDir, listener_child, networkFile) {
+        absConfigFile = require(util.resolvePath(networkFile));
+        if (this.config.hasOwnProperty('WITH_MQ') && this.config.WITH_MQ) {
+            clientUtil._consumeEvents(function(err){
+                logger.error(err);
+                listener_child.send({type:'closeKafkaProducer', config: absConfigFile});
+                clientUtil.stop();
+                clientUtil.closeKafkaConsumer();
+                demo.stopWatch();
+                if (absConfigFile.hasOwnProperty('caliper') && absConfigFile.caliper.hasOwnProperty('command') && absConfigFile.caliper.command.hasOwnProperty('start')) {
```

Comment:
When occurring an error, it should execute the  end command. So this line and next line should check the end property